site/app/templates/misc/Markdown.twig

```diff
@@ -0,0 +1,49 @@
+<div id="{{ id }}" class="markdown {{ class }}" style="{{ style ? style : '' }}">{{ content | markdown }}</div>
+<script>
+    $('.markdown code').parent().css('background-color', 'var(--standard-hover-light-gray');
+    $('.markdown code').parent().css('padding', '5px');
+    $('.markdown code').parent().css('white-space', 'pre');
+    $('.markdown code').parent().css('overflow-x', 'scroll');
+    $('.markdown code').parent().addClass('markdown-code');
+
+    $('.markdown code').each( function(index, code_area) {
+        const textarea = document.createElement('textarea');
+        //get the first matching group from this regex, or the from the backup array (plain)
+        const language = (code_area.className.match(/\blanguage-(.+)/) || ['', 'plain'])[1];
+
+        //match language to a CodeMirror mimeMode
+        let mime_matches = Object.keys(CodeMirror.mimeModes).filter(mimeMode => {
+            return (
+                mimeMode.indexOf(`text/x-${language}`) === 0 ||
+                mimeMode.indexOf(`text/${language}`) === 0 ||
+                mimeMode.indexOf(`application/x-${language}`) === 0 ||
+                mimeMode.indexOf(`application/${language}`) === 0
+            );
+        });
+
+        let mode;
+        if(!mime_matches.length){
+            mode = CodeMirror.mimeModes['text/plain'];
+        }
+        else {
+            mode = mime_matches[0];
+        }
+
+        textarea.innerHTML = code_area.innerHTML;
+        $(textarea).insertAfter(code_area);
+        $(code_area).remove();
+        const editor = CodeMirror.fromTextArea(textarea, {
+            lineNumbers: true,
+            theme: localStorage.theme ? (localStorage.theme === 'light' ? 'eclipse' : 'monokai') : 'eclipse',
+            readOnly: true,
+            mode: mode,
+        });
+
+        //set timeout to ensure refresh happens after new content has been set
+        //using 1 ms works, but 100 ms is used to be safe
+        setTimeout(function() {
+            editor.refresh();
+        },100);
+    });
+
+</script>
```

Comment:
It may be worth looking into switching from Parsedown to [League\CommonMark](https://commonmark.thephpleague.com/) as it would make possible to attach our own custom renderers for code blocks so that we don't have to do this kind of janky JS DOM manipulation post-hoc.

Reply:
Okay, I'll look into that. Do you think I should try to incorporate that into this PR or do it in a separate one?

Reply:
@MasterOdin I have everything working on the new engine except for line numbers in the code boxes. I've been at this for a few days now, but it seems some of the documentation for League\CommonMark is outdated or incomplete, making it very difficult to figure out what exactly I should be doing, and how I can get extensions (like the syntax highlighter) to work alongside my own since they both deal with rendering code blocks. 

I could probably do DOM manipulation to include the line numbers, but that would defeat the whole point of switching to this new markdown engine, unless you think it would be worth it to switch anyways since this engine is getting updated more. Any suggestions?

Reply:
Would you be able to post your work in a branch so I can see it?

Reply:
Sorry for the delayed response. My implementation is now in the `commonmark` branch.